src/trace/traceTypes.ts

```diff
@@ -77,14 +84,53 @@ export type ActionTraceEvent = {
   error?: string,
 };
 
+export type DialogOpenedEvent = {
+  timestamp: number,
+  type: 'dialog-opened',
+  contextId: string,
+  pageId: string,
+  dialogType: string,
+  message?: string,
+};
+
+export type DialogClosedEvent = {
+  timestamp: number,
+  type: 'dialog-closed',
+  contextId: string,
+  pageId: string,
+  dialogType: string,
+};
+
+export type NavigationEvent = {
+  timestamp: number,
+  type: 'navigation',
+  contextId: string,
+  pageId: string,
+  url: string,
+  sameDocument: boolean,
+};
+
+export type MiscEvent = {
```

Comment:
I'd rather these be split into standalone load and error events.

Reply:
Done.